packages/itwin/tree-widget/src/components/trees/experimental/common/UseHierarchyFiltering.tsx

```diff
@@ -0,0 +1,193 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useCallback, useMemo, useState } from "react";
+import { useDebouncedAsyncValue } from "@itwin/components-react";
+import { IModelConnection } from "@itwin/core-frontend";
+import { ClassInfo, DefaultContentDisplayTypes, Descriptor, KeySet } from "@itwin/presentation-common";
+import {
+  PresentationInstanceFilter, PresentationInstanceFilterDialog, PresentationInstanceFilterInfo, PresentationInstanceFilterPropertiesSource,
+} from "@itwin/presentation-components";
+import { Presentation } from "@itwin/presentation-frontend";
+import { GenericInstanceFilter, RowsLimitExceededError } from "@itwin/presentation-hierarchies";
+import { HierarchyLevelDetails, useTree } from "@itwin/presentation-hierarchies-react";
+import { InstanceKey } from "@itwin/presentation-shared";
+
+type UseTreeResult = ReturnType<typeof useTree>;
+
+interface UseHierarchyFilteringOwnProps {
+  imodel: IModelConnection;
+  defaultHierarchyLevelSizeLimit: number;
+}
+
+type UseHierarchyFilteringProps = UseHierarchyFilteringOwnProps & Pick<UseTreeResult, "getHierarchyLevelDetails">;
+
+/** @internal */
+export function useHierarchyFiltering({ imodel, defaultHierarchyLevelSizeLimit, getHierarchyLevelDetails }: UseHierarchyFilteringProps) {
```

Comment:
can we change the name of this to `useHierarchyLevelFiltering`?

Reply:
Renamed